File: Pulse/BasicPulsePlayer-iOS/objc/BasicPulsePlayer/BCOVPulseVideoItem.m
//
//  BCOVPulseVideoItem.m
//  BasicPulsePlayer
//
//  Copyright © 2024 Brightcove, Inc. All rights reserved.
//

#import "BCOVPulseVideoItem.h"


@implementation BCOVPulseVideoItem

+ (BCOVPulseVideoItem *)initWithDictionary:(NSDictionary *)dictionary
{
    BCOVPulseVideoItem *videoItem = [BCOVPulseVideoItem new];

    videoItem.title            = dictionary[@"content-title"];
    videoItem.category         = dictionary[@"category"];
    videoItem.tags             = dictionary[@"tags"];
    videoItem.midrollPositions = dictionary[@"midroll-positions"];
    videoItem.extendSession    = dictionary[@"extend-session"];

    return videoItem;
}

@end

File: Pulse/BasicPulsePlayer-iOS/BasicPulsePlayer/BCOVPulseVideoItem.swift
//
//  BCOVPulseVideoItem.swift
//  BasicPulsePlayer
//
//  Copyright © 2026 Brightcove, Inc. All rights reserved.
//

import Foundation


final class BCOVPulseVideoItem: NSObject {

    fileprivate(set) var title: String?
    fileprivate(set) var category: String?
    fileprivate(set) var tags: [String]?
    fileprivate(set) var midrollPositions: [NSNumber]?
    fileprivate(set) var extendSession: Bool?

    static func staticInit(dictionary: [String: Any]) -> BCOVPulseVideoItem {
        let videoItem = BCOVPulseVideoItem()

        videoItem.title = dictionary["content-title"] as? String ?? ""
        videoItem.category = dictionary["category"] as? String
        videoItem.tags = dictionary["tags"] as? [String]
        videoItem.midrollPositions = dictionary["midroll-positions"] as? [NSNumber]
        videoItem.extendSession = dictionary["extend-session"] as? Bool

        return videoItem
    }
}
